### PR TITLE
chore(rpc): improve debug_traceBlock performance

### PR DESCRIPTION
**Pull request by JosepBove**
Implements https://github.com/paradigmxyz/reth/issues/11066

### CODE REVIEW

**Comment by JosepBove** [comment]
First approach done, ready for review & feedback

File: crates/rpc/rpc/src/debug.rs [hunk 1]
```diff
@@ -332,7 +341,7 @@ where
                                 Ok(frame.into())
                             })
                             .await?;
-                        return Ok(frame)
+                        return Ok(frame);
```
Comment (mattsse) [inline crates/rpc/rpc/src/debug.rs:344]: could you please undo these, makes it harder to review, idk why they keep reappearing...

**Review by mattsse — COMMENTED** [review]
this is slightly more complex, and requires a different approach, maybe like an Optional Inspector as argument and the trace_transaction also returns the inspector again

File: crates/rpc/rpc/src/debug.rs (hunk 1 above)
Comment (emhane) [reply crates/rpc/rpc/src/debug.rs:344]: when you run `cargo fmt` and not `cargo +nightly fmt`...doom

Comment (JosepBove) [reply crates/rpc/rpc/src/debug.rs:344]: Oopsie, will update them

Comment (JosepBove) [reply crates/rpc/rpc/src/debug.rs:344]: About the approach does modifying the `trace_transaction` function to accept an optional inspector and return it along with the result and then modify `trace_block` function to use it sound good?

**Pull request closed without merging**